docs/operations-guide/environment-variables.md

```diff
@@ -0,0 +1,63 @@
+# Environment variables
+
+Settings in Metabase are created using a custom macro, `defsetting`. This macro helps to create a Settings model that will be cached and stored in the Database model, viewed/modified from the Admin Panel, and automagically creates a corresponding environment variable that can be used (by default) to override that setting.
+
+These environment variables, if defined, will prevent the corresponding setting from being changed in the Admin Panel.
```

Comment:
Is this intended to be a comprehensive list or just the ones defined with `defsetting`? There are some additional settings in `config.clj` and `plugins.clj` (not sure if others exist, I've just used `MB_DB_FILE` and `MB_PLUGIN_DIR` recently) - should we include those as well?

Reply:
You're right, there are some cases where we do `(config/config-str :mb-whatever)` without having a corresponding `defsetting`.

Here's a command to get a comprehensive list:

```
(grep -rEoh 'mb-[a-z-]+' src ; grep -rEoh 'defsetting [a-z-]+' src | awk '{ print "mb-" $2 }') | sort | uniq
```

Reply:
> You're right, there are some cases where we do `(config/config-str :mb-whatever)` without having a corresponding `defsetting`.

Are all those also exposed in the Admin Panel?

Reply:
Approving for now - there are 119 variables in that list and 55 `defsetting` ones. We can add the other ones as necessary given they may not have built in defaults or descriptions.